engines/wintermute/base/base_game.cpp

```diff
@@ -4249,6 +4344,40 @@ char *BaseGame::getKeyFromStringTable(const char *str) const {
 	return _settings->getKeyFromStringTable(str);
 }
 
+#ifdef ENABLE_HEROCRAFT
+uint8 BaseGame::getFilePartChecksumHc(const char *filename, size_t begin, size_t end) {
+	if (begin >= end) {
+		warning("Wrong limits for checksum check");
+		return 0;
+	}
+
+	uint32 size;
+	char *buffer = (char *)BaseFileManager::getEngineInstance()->readWholeFile(filename, &size);
+	if (buffer == nullptr) {
+		warning("Failed to open '%s' for checksum check", filename);
+		return 0;
+	}
+
+	if (size < end) {
+		warning("File '%s' is too small for checksum check", filename);
+		delete[] buffer;
+		return 0;
+	}
+
+	uint8 result = 0;
+	for (size_t i = begin; i < end; i++) {
```

Comment:
Please use ScummVM types, not `size_t`

Reply:
Fixed. But let me ask, what is wrong with `size_t`? It's widely used in tons of code, including all kinds of engines, backends and even `common/str.h`.